core/src/main/java/org/apache/iceberg/StaticTableScan.java

```diff
@@ -37,6 +38,28 @@ private StaticTableScan(TableOperations ops, Table table, Schema schema,
     this.buildTask = buildTask;
   }
 
+  /**
+   * Type of scan being performed by the buildTask, such as {@link MetadataTableType#HISTORY} when scanning
+   * a table's {@link org.apache.iceberg.HistoryTable}.
+   * <p>
+   * Used for logging and error messages.
+   */
+  protected String tableType() {
+    return "static";
+  }
+
+  @Override
+  public TableScan appendsBetween(long fromSnapshotId, long toSnapshotId) {
+    throw new UnsupportedOperationException(
+        String.format("Incremental scan is not supported for %s scan of table %s", tableType(), table().name()));
+  }
+
+  @Override
+  public TableScan appendsAfter(long fromSnapshotId) {
+    throw new UnsupportedOperationException(
+        String.format("Incremental scan is not supported for %s scan of table %s", tableType(), table().name()));
```

Comment:
This error message makes it sound like it is a combination of the table and the metadata table type, but no metadata tables support incremental scan (yet?).

I think this should state that the metadata table does not support incremental scan: `"Cannot incrementally scan metadata table: %s"`

Reply:
Correct. No metadata table supports incremental scan.

And it would be unlikely that they would as they all typically have snapshotId in them and time columns, though possibly as a convenience.

Reply:
I moved the argument to the constructor to get rid of the default of `"static"`.

As `BaseAllMetadataTable`'s `tableType` method is already abstract, I left it as is.

I changed the error message universally to `Cannot incrementally scan table of type $tableType`. Please let me know if there's anything else you'd like me to update.